pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,333 @@
+	// resource requirements for the sandbox are needed at creation time.
+	Resources PodSandboxResources
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+
+	// Linux contains configurations specific to Linux hosts.
+	Linux *LinuxPodSandboxConfig
+}
+
+// Labels are key value pairs that may be used to scope and select individual resources.
+// Label keys are of the form:
+//     label-key ::= prefixed-name | name
+//     prefixed-name ::= prefix '/' name
+//     prefix ::= DNS_SUBDOMAIN
+//     name ::= DNS_LABEL
+type Labels map[string]string
+
+// LinuxPodSandboxConfig holds platform-specific configuraions for Linux
+// host platforms and Linux-based containers.
+type LinuxPodSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sandbox's namespaces.
+	// This will be used only if the PodSandbox uses namespace for isolation.
+	NamespaceOptions NamespaceOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searches []string
+}
+
+type PodSandboxState string
+
+const (
+	// PodSandboxActive means the sandbox is functioning properly.
+	PodSandboxActive PodSandboxState = "active"
+	// PodSandboxInactive means the sandbox is not functioning properly.
+	PodSandboxInactive PodSandboxState = "inactive"
+)
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+type PodSandboxFilter struct {
+	// Name of the sandbox.
+	Name string
+	// ID of the sandbox.
+	ID PodSandboxID
+	// State of the sandbox.
+	State string
+	// LabelSelector to select matches.
+	// Only api.MatchLabels is supported for now and the requirements
+	// are ANDed. MatchExpressions is not supported yet.
+	LabelSelector unversioned.LabelSelector
+}
+
+// PodSandboxListItem contains minimal information about a sandbox.
+type PodSandboxListItem struct {
+	ID    string
+	State PodSandboxState
+}
+
+// PodSandboxStatus contains the status of the PodSandbox.
+type PodSandboxStatus struct {
+	// ID of the sandbox.
+	ID string
+	// State of the sandbox.
+	State PodSandboxState
+	// Network contains network status if network is handled by the runtime.
+	Network *PodSandboxNetworkStatus
+	// Status specific to a Linux sandbox.
+	Linux *LinuxPodSandboxStatus
+}
```

Comment:
@feiskyer, this may be good for pod-based runtimes such as hyper, but it's not the case for other runtimes. Having them separately simplifies the code in kubelet.